src/domain/request/requestExecutor/RequestExecutorFactory.ts

```diff
@@ -0,0 +1,35 @@
+import {LoggerPort} from '@core/domain/common/port';
+import FetchDataUseCase from '@core/domain/common/usecase/FetchDataUseCase';
+import SelectDataUseCase from '@core/domain/common/usecase/SelectDataUseCase';
+import RequestExecutor from '../../../domain/common/port/RequestExecutorPort';
+import InvalidContentTypeError from '../../../domain/common/utils/error/InvalidContentTypeError';
+import UrlRequestExecutor from './UrlRequestExecutor';
+
+class RequestExecutorFactory  {
+    private requestExecutors: RequestExecutor[];
+    constructor(
+        fetchDataUseCase: FetchDataUseCase,
+        selectDataUseCase: SelectDataUseCase,
+        logger: LoggerPort,
+    ) {
+        this.requestExecutors = [new UrlRequestExecutor(fetchDataUseCase, selectDataUseCase, logger)];
+    }
+
+    create(contentType: string): RequestExecutor {
+        let chosenRequestExecutor = null;
```

Comment:
sounds like find/some array method ;)

Reply:
`this.requestExecutors.find(re => re.canHandle(contentType))` :)